src/messenger.rs

```diff
@@ -6,57 +6,39 @@ use crate::slack::{self, SlackAttachment, SlackRequest};
 use crate::util;
 use crate::worker::Worker;
 
-pub trait Messenger {
-    fn send_to_all(
-        &self,
-        msg: &str,
-        attachments: &Vec<SlackAttachment>,
-        item_owner: &github::User,
-        sender: &github::User,
-        repo: &github::Repo,
-        participants: &Vec<github::User>,
-    );
-
-    fn send_to_owner(
-        &self,
-        msg: &str,
-        attachments: &Vec<SlackAttachment>,
-        item_owner: &github::User,
-        repo: &github::Repo,
-    );
-
-    fn send_to_channel(&self, msg: &str, attachments: &Vec<SlackAttachment>, repo: &github::Repo);
-}
-
-
-struct SlackMessenger {
-    pub config: Arc<Config>,
-    pub slack: Arc<dyn Worker<SlackRequest>>,
+pub struct Messenger {
+    config: Arc<Config>,
+    slack: Arc<dyn Worker<SlackRequest>>,
 }
 
-pub fn new(config: Arc<Config>, slack: Arc<dyn Worker<SlackRequest>>) -> impl Messenger {
-    SlackMessenger {
+pub fn new(config: Arc<Config>, slack: Arc<dyn Worker<SlackRequest>>) -> Messenger {
+    Messenger {
         slack: slack.clone(),
         config: config.clone(),
     }
 }
 
-impl Messenger for SlackMessenger {
-    fn send_to_all(
+impl Messenger {
+    pub fn send_to_all<T: github::CommitLike>(
```

Comment:
This was sorta a sad trade-off -- lose the `trait` to be able to get templated parameters. In the end, I liked the templated params more than the trait. 🤷‍♂️

Reply:
is the Trait `SlackMessenger` ?

Reply:
`Messenger` was the trait. traits don't work w/ generic/template functions. So, dropped the trait and just exposed the `SlackMessenger` type directly now as `Messenger`